arkouda/groupbyclass.py

```diff
@@ -990,6 +1003,382 @@ def broadcast(self, values : pdarray, permute : bool=True) -> pdarray:
+        they are unregistered.
+        """
+
+        if not self.name:
+            raise RegistrationError("This item does not have a name and does not appear to be registered.")
+
+        # Unregister all components in keys in the case of a Sequence
+        if isinstance(self.keys, Sequence):
+            for x in range(len(self.keys)):
+                self.keys[x].unregister()
+                self.unique_keys[x].unregister()
+        else:
+            from arkouda import Categorical
+            self.keys.unregister()
+
+            if not isinstance(self.unique_keys, Categorical):
+                self.unique_keys.unregister()
+
+        self.permutation.unregister()
+        self.segments.unregister()
+
+        self.name = None  # Clear our internal GroupBy object name
+
+    def is_registered(self) -> bool:
+        """
+         Return True iff the object is contained in the registry
+
+        Returns
+        -------
+        bool
+            Indicates if the object is contained in the registry
+
+        Raises
+        ------
+        RegistrationError
+            Raised if there's a server-side error or a mis-match of registered components
+
+        See Also
+        --------
+        register, attach, unregister, unregister_groupby_by_name
+
+        Notes
+        -----
+        Objects registered with the server are immune to deletion until
+        they are unregistered.
+        """
+        from arkouda import Categorical
+
+        if self.name is None:
+            return False  # unnamed GroupBy cannot be registered
+
+        if isinstance(self.keys, Sequence):  # Sequence - Check for all components
+            from re import match, compile
+            # Only check for a single component of Categorical to ensure correct count.
+            regEx = compile(f"^\\d+_{self.name}_.+\\.keys$|^\\d+_{self.name}_.+\\.unique_keys$|"
+                            f"^\\d+_{self.name}_.+\\.unique_keys(?=\\.categories$)")
+            cat_regEx = compile(f"^\\d+_{self.name}_categorical\\.keys(?=\\.codes$)")
+            simple_registered = list(filter(regEx.match, list_registry()))
+            cat_registered = list(filter(cat_regEx.match, list_registry()))
+            if f"{self.name}.permutation" in list_registry():
+                simple_registered.append(f"{self.name}.permutation")
+            if f"{self.name}.segments" in list_registry():
+                simple_registered.append(f"{self.name}.segments")
+
+            # In the case of Categorical, unique keys is registered with only categories and codes and
+            # overwrites keys.categories
+            total = (len(self.keys) * 2) + 2
+
+            registered = len(simple_registered) + len(cat_registered)
+            if 0 < registered < total:
+                print(f"WARNING: GroupBy {self.name} expected {total} components to be registered,"
+                      f" but only located {registered}. Found components have been unregistered.")
```

Comment:
Updated to use `warnings` instead of printing an error. After talking with @pierce314159 we are in agreement that we should probably update `Categorical` to use `warnings` here and return `false` instead of throwing an error.